src/workerd/api/node/tests/zlib-nodejs-test.js

```diff
@@ -213,3 +214,117 @@ export const crc32Test = {
+  "GUNZIP": 4,
+  "DEFLATERAW": 5,
+  "INFLATERAW": 6,
+  "UNZIP": 7,
+  "BROTLI_DECODE": 8,
+  "BROTLI_ENCODE": 9,
+  "Z_MIN_WINDOWBITS": 8,
+  "Z_MAX_WINDOWBITS": 15,
+  "Z_DEFAULT_WINDOWBITS": 15,
+  "Z_MIN_CHUNK": 64,
+  "Z_MAX_CHUNK": null,
+  "Z_DEFAULT_CHUNK": 16384,
+  "Z_MIN_MEMLEVEL": 1,
+  "Z_MAX_MEMLEVEL": 9,
+  "Z_DEFAULT_MEMLEVEL": 8,
+  "Z_MIN_LEVEL": -1,
+  "Z_MAX_LEVEL": 9,
+  "Z_DEFAULT_LEVEL": -1,
+  "BROTLI_OPERATION_PROCESS": 0,
+  "BROTLI_OPERATION_FLUSH": 1,
+  "BROTLI_OPERATION_FINISH": 2,
+  "BROTLI_OPERATION_EMIT_METADATA": 3,
+  "BROTLI_PARAM_MODE": 0,
+  "BROTLI_MODE_GENERIC": 0,
+  "BROTLI_MODE_TEXT": 1,
+  "BROTLI_MODE_FONT": 2,
+  "BROTLI_DEFAULT_MODE": 0,
+  "BROTLI_PARAM_QUALITY": 1,
+  "BROTLI_MIN_QUALITY": 0,
+  "BROTLI_MAX_QUALITY": 11,
+  "BROTLI_DEFAULT_QUALITY": 11,
+  "BROTLI_PARAM_LGWIN": 2,
+  "BROTLI_MIN_WINDOW_BITS": 10,
+  "BROTLI_MAX_WINDOW_BITS": 24,
+  "BROTLI_LARGE_MAX_WINDOW_BITS": 30,
+  "BROTLI_DEFAULT_WINDOW": 22,
+  "BROTLI_PARAM_LGBLOCK": 3,
+  "BROTLI_MIN_INPUT_BLOCK_BITS": 16,
+  "BROTLI_MAX_INPUT_BLOCK_BITS": 24,
+  "BROTLI_PARAM_DISABLE_LITERAL_CONTEXT_MODELING": 4,
+  "BROTLI_PARAM_SIZE_HINT": 5,
+  "BROTLI_PARAM_LARGE_WINDOW": 6,
+  "BROTLI_PARAM_NPOSTFIX": 7,
+  "BROTLI_PARAM_NDIRECT": 8,
+  "BROTLI_DECODER_RESULT_ERROR": 0,
+  "BROTLI_DECODER_RESULT_SUCCESS": 1,
+  "BROTLI_DECODER_RESULT_NEEDS_MORE_INPUT": 2,
+  "BROTLI_DECODER_RESULT_NEEDS_MORE_OUTPUT": 3,
+  "BROTLI_DECODER_PARAM_DISABLE_RING_BUFFER_REALLOCATION": 0,
+  "BROTLI_DECODER_PARAM_LARGE_WINDOW": 1,
+  "BROTLI_DECODER_NO_ERROR": 0,
+  "BROTLI_DECODER_SUCCESS": 1,
+  "BROTLI_DECODER_NEEDS_MORE_INPUT": 2,
+  "BROTLI_DECODER_NEEDS_MORE_OUTPUT": 3,
+  "BROTLI_DECODER_ERROR_FORMAT_EXUBERANT_NIBBLE": -1,
+  "BROTLI_DECODER_ERROR_FORMAT_RESERVED": -2,
+  "BROTLI_DECODER_ERROR_FORMAT_EXUBERANT_META_NIBBLE": -3,
+  "BROTLI_DECODER_ERROR_FORMAT_SIMPLE_HUFFMAN_ALPHABET": -4,
+  "BROTLI_DECODER_ERROR_FORMAT_SIMPLE_HUFFMAN_SAME": -5,
+  "BROTLI_DECODER_ERROR_FORMAT_CL_SPACE": -6,
+  "BROTLI_DECODER_ERROR_FORMAT_HUFFMAN_SPACE": -7,
+  "BROTLI_DECODER_ERROR_FORMAT_CONTEXT_MAP_REPEAT": -8,
+  "BROTLI_DECODER_ERROR_FORMAT_BLOCK_LENGTH_1": -9,
+  "BROTLI_DECODER_ERROR_FORMAT_BLOCK_LENGTH_2": -10,
+  "BROTLI_DECODER_ERROR_FORMAT_TRANSFORM": -11,
+  "BROTLI_DECODER_ERROR_FORMAT_DICTIONARY": -12,
+  "BROTLI_DECODER_ERROR_FORMAT_WINDOW_BITS": -13,
+  "BROTLI_DECODER_ERROR_FORMAT_PADDING_1": -14,
+  "BROTLI_DECODER_ERROR_FORMAT_PADDING_2": -15,
+  "BROTLI_DECODER_ERROR_FORMAT_DISTANCE": -16,
+  "BROTLI_DECODER_ERROR_DICTIONARY_NOT_SET": -19,
+  "BROTLI_DECODER_ERROR_INVALID_ARGUMENTS": -20,
+  "BROTLI_DECODER_ERROR_ALLOC_CONTEXT_MODES": -21,
+  "BROTLI_DECODER_ERROR_ALLOC_TREE_GROUPS": -22,
+  "BROTLI_DECODER_ERROR_ALLOC_CONTEXT_MAP": -25,
+  "BROTLI_DECODER_ERROR_ALLOC_RING_BUFFER_1": -26,
+  "BROTLI_DECODER_ERROR_ALLOC_RING_BUFFER_2": -27,
+  "BROTLI_DECODER_ERROR_ALLOC_BLOCK_TYPE_TREES": -30,
+  "BROTLI_DECODER_ERROR_UNREACHABLE": -31
```

Comment:
nit: if constants are added later it would be easy for this list to get out of date. I wonder if it would make sense to have the test fail if there are any unexpected constants in the `constants` object. You can check for that fairly easily by grabbing the `Object.keys()` and removing each item that is explicitly tested here then throwing if there are any left over at the end.

Reply:
Are you suggesting to test for this list of keys, and not check the values, or am I misunderstanding the suggestion?

Reply:
For for the list of keys and the values, but throw if there are additional keys in `constants` not tested for explicitly.